db/memtable.cc

```diff
@@ -783,6 +790,10 @@ Status MemTable::Add(SequenceNumber s, ValueType type,
         type == kTypeDeletionWithTimestamp) {
       num_deletes_.store(num_deletes_.load(std::memory_order_relaxed) + 1,
                          std::memory_order_relaxed);
+    } else if (type == kTypeRangeDeletion) {
```

Comment:
Could we set `max_tombstones_reached_ ` here? It would allow the memtable to be flushed without a following read.

Reply:
My particular use case is read+range-delete, so this suffices for me.
I found that a write + range-delete workload is adversely affected by moving this check out of Read path

Can we find a middle ground ? 

Let me discuss with team to understand if we moving this out adversely affects our use case.

Reply:
The flag will be set in range-deletions, as requested.
Looks like our use case  will work well with it.